### PR TITLE
Sreeraj/feature:  added productwise report

### PR DESCRIPTION
**Pull request by djsreeraj**
@kirananto ( 2 errors )  with useEffect dependencies,  I tried different ways when productData is made as dependency the count going infinite.

### CODE REVIEW

**Comment by kirananto** [comment]
Conflicting, also there'll be other approaches lemme go through the code first

**Comment by djsreeraj** [comment]
@kirananto  I have fixed the conflicts . Only the dependency error left

File: src/Pages/Reports/ProductReports/ProductReports.tsx [hunk 1]
```diff
@@ -1,35 +1,175 @@
...
+                if(result?.data?.total === 0){
+                    setLoading(false)
+                    setIsEmptyList(true)
+                }
+            })
+        }
+    }, [token])
+
+    useEffect(() => {
+        setOrderItems([])
```
Comment (kirananto) [inline src/Pages/Reports/ProductReports/ProductReports.tsx]: Wouldn't this conflict with the firstUseEffect ? We are not having a solid lock on the order of execution, instead relying on the delay which is variable.

File: src/Pages/Reports/ProductReports/ProductReports.tsx [hunk 2]
```diff
@@ -1,35 +1,175 @@
...
+        setProductData([])
+        getOrderItems()
```
Comment (kirananto) [inline src/Pages/Reports/ProductReports/ProductReports.tsx]: What does this useEffect Actually do ? I think that this is unnecessary complexity induced over here.

File: src/Pages/Reports/ProductReports/ProductReports.tsx [hunk 3]
```diff
@@ -1,35 +1,175 @@
...
+    }, [orders])
+
+
+    useEffect(() => {
+        getProductWiseData()
+    }, [orderItems])
+
+    const getOrderItems = () => {
+        orders?.map?.((order : orderInterface) => {
+            fetchOrderItems({ token, orderId: order.id, limit: 100, offset: 0 }).then((result) => {
```
Comment (kirananto) [inline src/Pages/Reports/ProductReports/ProductReports.tsx]: I'll rewrite this to a simplified approach, this is not readable

**Comment by kirananto** [comment]
Using states is expensive, you can always compute something and commit it for render phase, which is always more efficient than constantly updating the state.

https://blog.logrocket.com/why-react-doesnt-update-state-immediately/
https://www.reddit.com/r/reactjs/comments/aqv4np/how_expensive_is_a_react_rerender/

File: src/Pages/Reports/ProductReports/ProductReports.tsx [hunk 4]
```diff
@@ -1,35 +1,136 @@
...
+                    })
+                } else {
+                    _productData[index].productQty = _productData[index].productQty + item?.quantity
+                }
+            })
+            setProductData(_productData.sort?.((a, b) => b?.productQty - a?.productQty))
+        } catch (error) {
+            console.log(error)
+        }
+    }, [token])
```
Comment (kirananto) [inline src/Pages/Reports/ProductReports/ProductReports.tsx:59]: @djsreeraj  Please refer to this implementation

**Comment by kirananto** [comment]
> **Size Change:** +5.44 kB (+1%)
> 
> **Total Size:** 812 kB
> Filename 	Size 	Change 	
> `build/dist/Pages/Reports/ProductReports/ProductReports.js` 	1.59 kB 	+1.11 kB (+232%) 	🆘
> `build/dist/Pages/Signup/SignupStepThree.js` 	2.06 kB 	+6 B (0%) 	
> `build/dist/Pages/Signup/SignupStepTwo.js` 	1.94 kB 	+21 B (+1%) 	
> `build/dist/tailwind.css.proxy.js` 	10.2 kB 	+65 B (+1%) 	
> `build/js/194.js` 	0 B 	-437 B (removed) 	🏆
> `build/js/387.js` 	1.6 kB 	+8 B (+1%) 	
> `build/js/641.js` 	2.58 kB 	+6 B (0%) 	
> `build/js/dist/index.js` 	80.5 kB 	+5 B (0%) 	
> `build/sw.js` 	9.25 kB 	-4 B (0%) 	
> `build/dist/Pages/Reports/ProductReports/PrintAll.js` 	1.49 kB 	+1.49 kB (new file) 	🆕
> `build/dist/Pages/Reports/ProductReports/ProductCard.js` 	653 B 	+653 B (new file) 	🆕
> `build/js/311.js` 	2.51 kB 	+2.51 kB (new file) 	🆕
> ℹ️ **View Unchanged**
> 
> [compressed-size-action](https://github.com/preactjs/compressed-size-action)

Also try to see if we can reduce the size inflation here, later

**Pull request merged**